Static frontend

381e35 opened the pull request:
This pull request focuses on getting the front-end ready with all available static information.

masonchain committed:
Base set up

masonchain committed:
System message component

masonchain committed:
Navbar styled

masonchain committed:
Navbar finalized for MVP

Navbar functionality should be good to go, barring any slight style updates that come up after staring at it for a while.

masonchain committed:
Updating build config

masonchain committed:
Updating dependency path

masonchain committed:
Updating component import

masonchain committed:
Styling inheritance fixes

masonchain committed:
Footer good to go

381e35 committed:
updating working directory

381e35 committed:
updating jsconfig

381e35 committed:
Delete frontend/src/Components directory

masonchain committed:
Revert "Delete frontend/src/Components directory"

This reverts commit e10bb56c7c895578ecd9b5f60bdafcd90cc58062.

masonchain committed:
Merge branch '7-mobile-first-frontend-experience-app' of https://github.com/nftchance/runner into 7-mobile-first-frontend-experience-app

masonchain committed:
Renaming back to components

381e35 committed:
disabling eslint

masonchain committed:
Landing hero desktop styling

masonchain committed:
Second block of landing page

masonchain committed:
Landing page modules

masonchain committed:
Landing page done

masonchain committed:
Cleaning up components

masonchain committed:
Roadmap page created

The left margin indicator image is present, but this will probably be easier to properly handle not as a static image. On hold before I get to redoing all the static images in css.

masonchain committed:
FAQ YEAH

Still need to write the copy for the answers tho

masonchain committed:
Updated import paths, Loadable components

masonchain committed:
Governance page

This is mostly finalized, still need to implement the sort and some prep for the API integration, but had a lot of pending changes and wanted to push before heading out.

Also included is a lot of clean up with usage of single or double quotes. OCD but looks good

masonchain committed:
General clean up

masonchain committed:
Proposal page

masonchain committed:
Whole lotta style polishing

Just a couple of things left before API integration.

- Replacing bar chart and roadmap check images with css
- Responsiveness of all pages (this will probably be a bitch but I think I set myself up well, we'll see).
- Finally just copy and index.html finalization and then moving on to API int/account auth/dashboards

381e35 commented on .github/workflows/react.yml:
```diff
@@ -0,0 +1,44 @@
+    - run: |
+        npm config set '//npm.fontawesome.com/:_authToken' "${{ secrets.FONT_AWESOME_AUTH_TOKEN }}"
+        npm ci
+    - run: npm run build --if-present
```
Need to implement the ability to have automated testing to make sure that all of the pages are built properly and can be navigated to.

Just need to spin up each page and then have a request that validates the header. You could go crazy on this, but at this point we just need basic verification that everything is built and static-running fine.

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+☐ Mobile Responsiveness
```
If you keep your todo in the GitHub project, issues are automatically created. Can be assigned, and I can hop in and knock things out you haven't done. 

(I haven't been adding my small tasks because I haven't had a running list of things to do and instead have just been going sequentially)

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+    ☐ Responsivity of descriptions in cards
+        Not sure how to handle this just yet.
+        We need them to be the same height regardless of how many lines the description takes up
+        Give them a predefined height and text-overflow: ellipsis
```
The standard CSS way of doing it would be to make all cards inherit the row height and then make it just use auto height for the 'inline' card size.

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+    ☐ Awaiting websocket/REST data should render the components, just with empty data.
+    ☐ Governance page proposal list view
+    ☐ Should the API url go in a .env or is it okay in the constants file?
+    ☐ Email list integration
```
I haven't implemented any part of this. Will you please create an issue and spec this out. I did already handle the email collection and validation of users however there is no connection to that data at this time.

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+☐ API Integration:
+    ☐ Awaiting websocket/REST data should render the components, just with empty data.
+    ☐ Governance page proposal list view
+    ☐ Should the API url go in a .env or is it okay in the constants file?
```
Constants file is fine 👍

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+    ✔ Dynamic versioning image on margin @done(22-08-05 10:12)
+    ✔ Cards @done(22-08-05 10:12)
+    ✔ Status indicators @done(22-08-05 10:12)
+    ☐ Left aligned indicator put into CSS
```
This is a good example of something I could do if your items were in GitHub. I will go ahead and take care of this, but you could have just assigned this to me and I would've taken care of it for ya.

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+    ✔ Proposal Page @done(22-08-09 16:23)
+    ✔ Ellipsis Menu Items and their functions @done(22-08-10 02:22)
+    ☐ Finalize what the menu items should be and do
+    ☐ What should the meta description be? Just general for all proposals? Or contain the summary?
```
obj.summary

381e35 commented on frontend/TODO:
```diff
@@ -0,0 +1,152 @@
+☐ DASHBOARDS:
+
+☐ OTHER:
+    ☐ Opengraph images
```
Tricky. Not something I had thought about yet. Will think on this today so that we can add this before we have too many pages that we have to go back and update.

381e35 commented on frontend/src/App.css:
```diff
@@ -1,38 +1,185 @@
-  text-align: center;
+@import url("https://use.typekit.net/evr1bsi.css");
+
+:root {
```
Very happy you did it this way -- I think we are going to rework our color schema but I will let you know when that happens.

381e35 replied:
The weighting of color saturations is incorrect/illogical in some places. Thing we will be moving to base level color with hex and then tiering each step into:

- rgba(..., .15)
- rgba(..., .45)
- rgba(..., .65)
- rgba(..., .85)

But I will write usage docs and get back to this.

381e35 commented on frontend/src/components/Accordion/AccordionPanel.css:
```diff
@@ -0,0 +1,31 @@
+
+.expand-icon {
+    color: #FFFFFF;
+    opacity: 0.65;
```
Instead of changing opacity here the recommended way of doing this would be changing the color to `rgba(255,255,255,0.65);` since some browsers are absolutely fucked when it comes to opacity.

masonchain replied:
10/4, added replacing all opacity usage with color alpha layering. Is there any potential issue you can think of with using hex with the alpha value? (e.g. --dark-border-65: #3D3D3D65)?

381e35 replied:
I have legit never seen this so I'd say best to avoid lol

381e35 commented on frontend/src/components/Buttons/MenuButton.js:
```diff
@@ -0,0 +1,53 @@
+
+const MenuButton = ({icon, menuItems, handleMenu}) => {
+    const [ anchorEl, setAnchorEl ] = useState(null);
+    const open = anchorEl ? true : false;
```
Should be wrapped in `useMemo()`

381e35 commented on frontend/src/components/Buttons/MenuButton.js:
```diff
@@ -0,0 +1,53 @@
+            <IconButton 
+                className="menu-btn"
+                id="menu-btn"
+                aria-controls={open ? "menu" : undefined}
```
I'm not 100% sure -- but I think you can just do `open && "menu"` to save yourself from always needing to type `: undefined`.

381e35 commented on frontend/src/components/Buttons/SecondaryButton.js:
```diff
@@ -0,0 +1,21 @@
+            <Button
+                className="btn-secondary"
+                onClick={onClick}
+                disabled={disabled ? true : false}
```
Can just be `disabled={disabled}` or `disabled={disabled === 'string'}` if it is not a boolean being passed in.

381e35 replied:
Functionally no different, but much more readable.

381e35 commented on frontend/src/components/Cards/GhostCard.js:
```diff
@@ -0,0 +1,43 @@
+    return (
+        <div className="ghost-card">
+            <div className="title-line">
+                {noIcon ?
```
Can be changed to `{!noIcon && ...}`

381e35 commented on frontend/src/components/Cards/ProposalCard.js:
```diff
@@ -0,0 +1,54 @@
+            <div className="votes-bar">
+                <div 
+                    className={`yes-votes ${votingPercents[0] === 0 ? 'hidden' : ''}`}
+                    style={{width: `${votingPercents[0]}%`}}/>
```
Right now proposal vote %s are not exposed in the API. While total amounts are, we do technically have that data that can be calculated on the front-end but I would prefer to deliver that already calculated information if we plan on using it.

Please confirm for me that I need to add %s to the API.

masonchain replied:
Percentages in the API would be great. The XD design calls for it in the proposal card and it is necessary for the width of the bars showing voting both on the proposal page and proposal cards on the governance page.

381e35 replied:
Has been added to the API. Returns as `vote_percentages` in a `dict` of `vote_type` like:

`{'for' Decimal('100'), 'against': Decimal('0.0000'), 'abstain': Decimal('0.0000')}`

381e35 commented on frontend/src/components/Cards/ProposalCard.js:
```diff
@@ -0,0 +1,54 @@
+                    style={{width: `${votingPercents[1]}%`}}
+                />
+                <div
+                    className={`missing-votes ${votingPercents[0] + votingPercents[1] === 100 ? 'hidden' : ''}`} 
```
I can also add this to the API. Will you start a issue and build out a general spec sheet so that I can make sure you have everything you need here.

381e35 commented on frontend/src/components/Cards/ProposalCard.js:
```diff
@@ -0,0 +1,54 @@
+        <div className="proposal-card">
+            <div className="proposal-info">
+                <div className="proposal-tags">
+                    {tags.map((tag) => (
```
Should tags be clickable so that you can filter down to all items of a given tag?

masonchain replied:
Yeah that'd be some nice UX. Adding it to the API integration issue I am about to create for ya.

381e35 commented on frontend/src/components/Governance/Governance.js:
```diff
@@ -0,0 +1,144 @@
+    const [ proposalData, setProposalData ] = useState();
+
+    // These are just placeholder, not sure what sort methods we want to provide or what the field will actually be.
+    const sortingMethods = [
```
This is also not supported in the API currently. Very easy to add, you can make the decision of what filtering makes sense. Just spec it out for me please.

381e35 commented on frontend/src/components/Governance/Governance.js:
```diff
@@ -0,0 +1,144 @@
+        },
+    ]
+
+    const filterMethods = [
```
This is very nice, but I forgot to implement this into the API. This will be taken care of without the need for you to spec it.

381e35 commented on frontend/src/components/Governance/Proposal.js:
```diff
@@ -0,0 +1,239 @@
+        'Open': 'green',
+    }
+
+    const votePercentages = [
```
I don't want to make you do it like this :( I will update the API. That's my bad.

381e35 commented on frontend/src/components/Governance/Proposal.js:
```diff
@@ -0,0 +1,239 @@
+                    <meta name="og:title" content={`runner | [RP${params.id}]`} />
+                    <meta name="og:description" content={`runner | [RP${params.id}]`} />
+
+                    {/* TODO: Figure out if the description is general or the summary proposal
```
Summary is for META and quick recap. Summary will only be used as an "additional" field when we've already shown all other info or don't have the room to do so.

381e35 commented on frontend/src/components/Governance/Proposal.js:
```diff
@@ -0,0 +1,239 @@
+                            ))}
+                        </div>
+                        <div className="proposal-title">
+                            <h1>{`[RP${params.id}] ${proposalData?.title}`}</h1>
```
I built `__str__()` for everything in the database so that you don't have to do this. Will figure out how to get this exposed in the API for everything.

381e35 commented on frontend/src/components/Governance/Proposal.js:
```diff
@@ -0,0 +1,239 @@
+                            </div>
+                            <div className="voting-bar">
+                                <div 
+                                    className={`voting-progress ${proposalData?.votes_against === 0 ? 'hidden' : ''}`} 
```
Can be written like:
`className={'voting-progress ${proposalData?.votes_against === 0 && 'hidden'}'}`

or even

`className={'voting-progress ${!proposalData?.votes_against && 'hidden'}'}` since 0 is falsy in Javascript.

381e35 commented on frontend/src/index.js:
```diff
@@ -1,14 +1,13 @@
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
+  // <React.StrictMode>
```
Please re-enable.

381e35 reviewed (commented):
Things overall look very nice 🚀 

Overall no significant thoughts on the architecture. It is precisely how I would've done it. I did have a few small things that will save you some time in the future and make things, in general, more readable that are just general syntax things that aren't a biggy.

The one thing that did stand out to me continually was that we need to include `StyledEngineProvider` in every component. Can that not be a wrapper? Would love it if we could find a way to avoid needing to include that every time -- If it can't be a wrapper, maybe it should just be a different ComponentType that you can inherit from as the parent.

In terms of things that I may need from you moving forward they are:

1. Please specify the things requested, and I will make sure to add those to the API so that you can implement all data pieces of the static front-end at once instead of waiting for me when it's 80% implemented.

2. For todo items, if you would use the GitHub project (I think you have access?) -- With that, I can hop into any PR that you have open and plug in something that you need my help with instead of needing to live share or catch each other actively available to work on the same thing.

Overall, excellent shit ser I am excited to see things get plugged in so we can see this puppy live. Things are coming together much faster than I expected.

masonchain committed:
Implement review comments

All comments that could be taken care of, were implemented. Still a few outstanding things with API integration but I'm already way out of scope of the name of this PR and branch lmao.

masonchain committed:
lord finally got test environment working

Pain in the ass to get everything resolving but its ready to go and can be fleshed out now.

This commit just includes checking the render of every current static page.

masonchain committed:
Volume Earning Graph in CSS

masonchain committed:
some :pinched_fingers: UX on the graph

masonchain committed:
Fixed hover overlapping with legend

masonchain committed:
FAQ Responsiveness

masonchain committed:
some changes that have been sitting here for a min
